### ISSUE TITLE
llvm-bolt --help is missing info about -report-bad-layout option

### ISSUE THREAD

**Issue by tvondra** [llvm/llvm-project#114230]
It seems `llvm-bolt --help` says nothing about `-report-bad-layout`. I only noticed because the BOLT paper mentions the option without the required uint argument, so the command was failing for me. In the end I found `bolt/docs/CommandLineArgumentReference.md` but `--help` doesn't mention the option at all.

**Comment by marcauberer** [comment]
The respective command line option is explicitly marked as hidden. See here:

https://github.com/llvm/llvm-project/blob/56dcfbef453d6cc390fc7a734db417e047616526/bolt/lib/Passes/BinaryPasses.cpp#L178

@maksfb is this intended?

**Comment by maksfb** [comment]
> @maksfb is this intended?

Yes. We don't have a strong definition of the "bad" layout and this option was indeed added for the paper to easily diagnose places where the compiler can generate suboptimal layout even in the presence of a high-quality profile. It was never intended for the end user. But now you seem to find a use for it, although I'm not convinced that's really what you need in the end.
